### PR TITLE
K19DKPA-151

### PR DESCRIPTION
**Pull request by sottiletta**

### CODE REVIEW

File: backend/src/main/java/edu/netcracker/backend/dao/mapper/TripMapper.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    private final TripStateRegistry tripStateRegistry;
+
+    @Autowired
+    public TripMapper(TripStateRegistry tripStateRegistry) {
+        this.tripStateRegistry = tripStateRegistry;
+    }
+
+    @Override
+    public Trip mapRow(ResultSet rs, int rowNum) throws SQLException {
```
Comment (EugeneLokotariev) [inline backend/src/main/java/edu/netcracker/backend/dao/mapper/TripMapper.java]: Operations with user repeat in this method. I suggest creating helper method in order to not to duplicate code

File: backend/src/main/java/edu/netcracker/backend/model/state/trip/Archived.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import edu.netcracker.backend.model.User;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.List;
+
+@Component
+public class Archived extends TripState {
+
+    private final static int value = 5;
```
Comment (EugeneLokotariev) [inline backend/src/main/java/edu/netcracker/backend/model/state/trip/Archived.java]: It's difficult to understand what this variable does. I suggest changing name. In other cases below all the same

File: backend/src/main/java/edu/netcracker/backend/model/state/trip/TripStateRegistry.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    private final Assigned assigned;
+    private final Published published;
+    private final Archived archived;
+    private final UnderClarification underClarification;
+    private final Removed removed;
+
+    private HashMap<Integer, TripState> registry;
+
+    @Autowired
+    public TripStateRegistry(Draft draft, Open open, Assigned assigned, Published published, Archived archived,
```
Comment (EugeneLokotariev) [inline backend/src/main/java/edu/netcracker/backend/model/state/trip/TripStateRegistry.java]: It seems this method haven't been changed to our code style

File: backend/src/main/java/edu/netcracker/backend/security/JwtAuthFilter.java [hunk 4]
```diff
@@ -77,41 +57,50 @@ protected void doFilterInternal(HttpServletRequest httpServletRequest,
...
         User user = userService.findByUsername(jwtProvider.retrieveSubject(refreshToken));
-        if (isNotMatchedWithUsersRefreshToken(refreshToken, user)) return null;
+        if (isNotMatchedWithUsersRefreshToken(refreshToken, user)) {
+            return null;
+        }
 
         return jwtProvider.generateAccessToken(user);
     }
 
     private void handleToken(String accessToken, HttpServletRequest request) {
```
Comment (EugeneLokotariev) [inline backend/src/main/java/edu/netcracker/backend/security/JwtAuthFilter.java]: In this method you should manually delete redundant lines

File: backend/src/main/java/edu/netcracker/backend/dao/mapper/TripMapper.java (hunk 1 above)
Comment (sottiletta) [reply backend/src/main/java/edu/netcracker/backend/dao/mapper/TripMapper.java]: Moved repeating logic to mapRow method.

File: backend/src/main/java/edu/netcracker/backend/model/state/trip/Archived.java (hunk 2 above)
Comment (sottiletta) [reply backend/src/main/java/edu/netcracker/backend/model/state/trip/Archived.java]: Renamed to databaseValue.

File: backend/src/main/java/edu/netcracker/backend/security/JwtAuthFilter.java (hunk 4 above)
Comment (sottiletta) [reply backend/src/main/java/edu/netcracker/backend/security/JwtAuthFilter.java]: Deleted to match code style.

**Pull request merged**